pkg/apps/controller/deployer/deployer_controller_test.go

```diff
@@ -1075,3 +1082,75 @@ func TestDeployerCustomLabelsAndAnnotations(t *testing.T) {
 		expectMapContains(t, podTemplate.Annotations, test.annotations, "annotations")
 	}
 }
+
+func TestMakeDeployerPod(t *testing.T) {
+	client := &fake.Clientset{}
+	controller := okDeploymentController(client, nil, nil, true, v1.PodUnknown)
+	config := deploytest.OkDeploymentConfig(1)
+	deployment, _ := deployutil.MakeDeploymentV1(config, codec)
+	container := controller.makeDeployerContainer(&config.Spec.Strategy)
+	container.Resources = deployutil.CopyApiResourcesToV1Resources(&config.Spec.Strategy.Resources)
+	defaultGracePeriod := int64(10)
+	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
+
+	for i := 1; i <= 25; i++ {
+		seed := rand.Int63()
+		f := fuzzer.FuzzerFor(kapitesting.FuzzerFuncs, rand.NewSource(seed), kapi.Codecs)
+		f.Funcs(
+			func(p *kapiv1.PodTemplateSpec, c fuzz.Continue) {
+				c.FuzzNoCustom(p)
+				p.Spec.InitContainers = nil
+
+				// These are specific for deployer pod container:
+				p.Spec.Containers = []kapiv1.Container{*container}
+				p.Spec.Containers[0].Name = "deployment"
+				p.Spec.Containers[0].Command = container.Command
+				p.Spec.Containers[0].Args = container.Args
+				p.Spec.Containers[0].Image = container.Image
+				p.Spec.Containers[0].Env = append(p.Spec.Containers[0].Env, kapiv1.EnvVar{Name: "OPENSHIFT_DEPLOYMENT_NAME", Value: "config-1"})
+				p.Spec.Containers[0].Env = append(p.Spec.Containers[0].Env, kapiv1.EnvVar{Name: "OPENSHIFT_DEPLOYMENT_NAMESPACE", Value: "default"})
+				p.Spec.Containers[0].Resources = container.Resources
+				p.Spec.Containers[0].ImagePullPolicy = kapiv1.PullIfNotPresent
+
+				// These are hardcoded for deployer pod spec
+				p.Spec.RestartPolicy = kapiv1.RestartPolicyNever
+				p.Spec.TerminationGracePeriodSeconds = &defaultGracePeriod
+				p.Spec.ActiveDeadlineSeconds = &maxDeploymentDurationSeconds
+				p.Spec.ServiceAccountName = "sa:test"
+
+				// FIXME: These are weird or missing. If you get an error below, consider
+				// adding this field into deployer controller or to this list:
+				p.Spec.DeprecatedServiceAccount = ""
+				p.Spec.AutomountServiceAccountToken = nil
+				p.Spec.Volumes = nil
+				p.Spec.NodeName = ""
+				p.Spec.HostNetwork = false
+				p.Spec.HostPID = false
+				p.Spec.HostIPC = false
+				p.Spec.Hostname = ""
+				p.Spec.Subdomain = ""
+				p.Spec.Affinity = nil
+				p.Spec.SchedulerName = ""
+				p.Spec.HostAliases = nil
+				p.Spec.Priority = nil
+				p.Spec.PriorityClassName = ""
```

Comment:
@smarterclayton reminds me that we should maybe populate this to SYSTEM for deployer pods?